### PR TITLE
feat: add Capture counts to Grower Details dialog

### LINKED ISSUE #132: Grower Detail: Show number of captures
The Planter Detail dialog should show the number of captures by the planter in each of the following states:
- Approved
- Awaiting Verification
- Rejected

There should be some visual consistency here with the Verification Status labels that can be seen in the Capture Detail dialog:

Screenshot 2021-07-10 at 17 26 10
Screenshot 2021-07-10 at 17 25 48
Screenshot 2021-07-10 at 17 26 26
 

~~Clicking each number should show the corresponding captures in the Verify tool (by setting the Verification Status and Planter ID filters to the appropriate values).~~

### PR DESCRIPTION
**Pull request by tranquanghuy0801**
## Description

_[Add a description of the changes]_
Added the Capture counts to Grower Detail dialog.

**Issue(s) addressed**

- Resolves #132

**What kind of change(s) does this PR introduce?**

- [x] Enhancement
- [x] Bug fix
- [ ] Refactor

**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**
The current Grower Detail dialog does not have the associated Capture counts

**What is the new behavior?**
The current Grower Detail dialog will have the associated Capture counts

## Breaking change

**Does this PR introduce a breaking change?**
No

## Other useful information

Screenshot

image

### CODE REVIEW

File: src/components/GrowerDetail.js [hunk 1]
```diff
@@ -106,6 +140,23 @@ const GrowerDetail = (props) => {
                   }))
                   .filter((id) => id),
               );
+              let statusCount = emptyStatusCount;
+              registrations.map((reg) => {
```
Comment (nmcharlton) [inline src/components/GrowerDetail.js:144]: These stats are meant to be capture (tree) counts, not grower (planter) registration counts.

**Review by nmcharlton — CHANGES_REQUESTED** [review]
The UI looks great, but these aren't the right counts.
We need to query the API for each count against this grower.

File: src/components/GrowerDetail.js (hunk 1 above)
Comment (tranquanghuy0801) [reply src/components/GrowerDetail.js:144]: Thanks for the feedback. I will make the changes

**Comment by tranquanghuy0801** [comment]
@nmcharlton  I made the changes based on your suggestions. Please check my work, thanks

File: src/components/GrowerDetail.js [hunk 2]
```diff
@@ -115,6 +154,37 @@ const GrowerDetail = (props) => {
     // eslint-disable-next-line
   }, [growerId, growerContext.growers]);
 
+  useEffect(() => {
+    async function loadCaptures() {
+      if (growerId) {
+				setLoading(true);
+        let filter = new FilterModel();
+        filter.planterId = growerId;
+        const captures = await capturesContext.getAllCaptures({ filter });
```
Comment (nmcharlton) [inline src/components/GrowerDetail.js:163]: This is an extremely inefficient way of querying counts.
You don't need to get the capture records from the API, just the capture counts (see `treeTrackerApi:getCaptureCount()`.
You can do one query per verification state, setting the active/approved values as necessary.

File: src/views/GrowersView.js [hunk 3]
```diff
@@ -19,12 +20,14 @@ function GrowersView() {
       style={{ flexWrap: 'nowrap', height: '100%', overflow: 'hidden' }}
     >
       <GrowerProvider>
-        <SpeciesProvider>
-          <TagsProvider>
-            <GrowerFilterHeader />
-            <Growers />
-          </TagsProvider>
-        </SpeciesProvider>
+        <CapturesProvider>
```
Comment (nmcharlton) [inline src/views/GrowersView.js:23]: You don't need to go through the CapturesContext for this change – just call the appropriate treeTrackerApi method directly, or through GrowerContext.

**Review by nmcharlton — CHANGES_REQUESTED** [review]

**Review by nmcharlton — APPROVED** [review]
Fantastic work, thanks @tranquanghuy0801!

**Pull request merged**